lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -174,6 +174,198 @@
+      sample: true
+    ebs_optimized:
+      description: Indicates whether the instance is optimized for EBS I/O (true) or not (false).
+      returned: when I(state=present)
+      type: bool
+      sample: false
+    image_id:
+      description: The ID of the Amazon Machine Image used by the launch configuration.
+      returned: when I(state=present)
+      type: string
+      sample: ami-9be6f38c
+    instance_monitoring:
+      description: Indicates whether instances in this group are launched with detailed (true) or basic (false) monitoring.
+      returned: when I(state=present)
+      type: bool
+      sample: true
+    instance_profile_name:
+      description: The name or Amazon Resource Name (ARN) of the instance profile associated with the IAM role for the instance.
+      returned: when I(state=present)
+      type: string
+      sample: null
+    instance_type:
+      description: The instance type for the instances.
+      returned: when I(state=present)
+      type: string
+      sample: t1.micro
+    iops:
+      description: The number of I/O operations per second (IOPS) to provision for the volume.
+      returned: when I(state=present)
+      type: NoneType
+      sample: null
+    kernel_id:
+      description: The ID of the kernel associated with the AMI.
+      returned: when I(state=present)
+      type: string
+      sample: ''
+    key_name:
+      description: The name of the key pair.
+      returned: when I(state=present)
+      type: string
+      sample: testkey
+    launch_configuration_arn:
+      description: The Amazon Resource Name (ARN) of the launch configuration.
+      returned: when I(state=present)
+      type: string
+      sample: arn:aws:autoscaling:us-east-1:148830907657:launchConfiguration:888d9b58-d93a-40c4-90cf-759197a2621a:launchConfigurationName/launch_config_name
+    member:
+      description: ""
+      returned: when I(state=present)
+      type: string
+      sample: "\n      "
+    name:
+      description: The name of the launch configuration.
+      returned: when I(state=present)
+      type: string
+      sample: launch_config_name
+    ramdisk_id:
+      description: The ID of the RAM disk associated with the AMI.
+      returned: when I(state=present)
+      type: string
+      sample: ''
+    security_groups:
+      description: The security groups to associate with the instances.
+      returned: when I(state=present)
+      type: list
+      sample:
+      - sg-5e27db2f
+    spot_price:
+      description: The price to bid when launching Spot Instances.
+      returned: when I(state=present)
+      type: NoneType
+      sample: null
+    use_block_device_types:
+      description: Indicates whether to suppress a device mapping.
+      returned: when I(state=present)
+      type: bool
+      sample: false
+    user_data:
+      description: The user data available to the instances.
+      returned: when I(state=present)
+      type: string
+      sample: ''
+    volume_type:
```

Comment:
I created a couple of launch configs to verify this. But I never get `volume_type` or `delete_on_termination` info back in the returned values. Is it returned conditionally, i.e., not always returned even when `state=present`?

In fact, the `result.block_device_mappings` that is returned seems incomplete. Look at the code, it seems like we populate a lot of more info in that dictionary. Should I create a separate issue for that?

Reply:
You're right, looks like things are not being consistently returned... I created a basic launch config and then ran the [return_skeleton_generator](https://github.com/ansible/ansible/blob/devel/hacking/return_skeleton_generator.py) on it. I'll do that a couple more times with different options and try to find what's conditionally returned.

Reply:
@prasadkatti Actually. I wrote this on Friday and then forgot- I generated this running Ansible 2.3 (since the module is stableinterface it should be the same). So the reason yours is different is either because something is conditionally returned or because backwards compatibility has been broken.